packages/tpetra/core/src/Tpetra_Details_Behavior.cpp

```diff
@@ -22,6 +24,89 @@ namespace { // (anonymous)
     return s;
   }
 
+  template<class Func>
+  void
+  split(const std::string& s, Func f, const char sep=':')
+  {
+    typedef std::string::size_type size_type;
+    size_type cur_pos, last_pos=0, length=s.length();
+    while(last_pos < length + 1)
+    {
+      cur_pos = s.find_first_of(sep, last_pos);
+      if(cur_pos == std::string::npos)
+      {
+        cur_pos = length;
+      }
+      if(cur_pos!=last_pos) {
+        auto token = std::string(s.data()+last_pos, (size_type)cur_pos-last_pos);
+        f(token);
+      }
+      last_pos = cur_pos + 1;
+     }
+    return;
+  }
+
+  /// \brief Determine if environment variable is a boolean or not
+  ///
+  /// If the environment variable is "true", return 1
```

Comment:
An `enum` might be nicer here, but this is fine for now.

Reply:
`enum` where?